pkg/i2gw/ingress2gateway.go

```diff
@@ -120,16 +124,24 @@ func ConstructIngressesFromCluster(ctx context.Context, cl client.Client, ingres
 
 // constructProviders constructs a map of concrete Provider implementations
 // by their ProviderName.
-//
-// TODO (levikobi): Issue #45 - let users filter by provider name.
-func constructProviders(conf *ProviderConf) map[ProviderName]Provider {
+func constructProviders(conf *ProviderConf, providers []string) (map[ProviderName]Provider, error) {
+
+	supportedProviders := GetSupportedProviders()
+	for _, provider := range providers {
+		if !slices.Contains(supportedProviders, provider) {
+			return nil, fmt.Errorf("The provider %s is unsupported", provider)
+		}
+	}
+
 	providerByName := make(map[ProviderName]Provider, len(ProviderConstructorByName))
 
 	for name, newProviderFunc := range ProviderConstructorByName {
-		providerByName[name] = newProviderFunc(conf)
+		if slices.Contains(providers, string(name)) {
+			providerByName[name] = newProviderFunc(conf)
+		}
 	}
 
-	return providerByName
+	return providerByName, nil
```

Comment:
You can take advantage of the fact that ProviderConstructorByName is a map to avoid the O(n^2) runtime complexity, and to make the code a bit more concise

```suggestion
	providerByName := make(map[ProviderName]Provider, len(ProviderConstructorByName))

	for _, requestedProvider := range providers {
		requestedProviderName := ProviderName(requestedProvider)
		newProviderFunc, ok := ProviderConstructorByName[requestedProviderName]
		if !ok {
			return nil, fmt.Errorf("%s is not a supported provider", requestedProvider)
		}
		providerByName[requestedProviderName] = newProviderFunc(conf)
	}

	return providerByName, nil
```

Reply:
Greate suggestion! Thanks very much.